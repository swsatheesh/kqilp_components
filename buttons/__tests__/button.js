import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../button';

jest.unmock('../button');

describe('component', () => {
  describe('<Button />', () => {
    it('should render correctly', () => {
      const clickHander = jest.fn();
      const button = renderer.create(
        <Button onClick={clickHander}>click me</Button>
      ).toJSON();
      expect(button).toMatchSnapshot();

      // Click callback
      button.props.onClick();
      expect(clickHander).toBeCalled();
      expect(button.props.className).toEqual('btn btn-outline init-caps');
      expect(button.children).toContain('click me');
    });

    it('should have a large size', () => {
      const button = renderer.create(
        <Button type={'solid'} size={'btn-lg'}>click me large</Button>
      ).toJSON();
      expect(button).toMatchSnapshot();
      expect(button.props.className).toEqual('btn btn-solid btn-lg init-caps');
    });

    it('should have a small size', () => {
      const button = renderer.create(
        <Button type={'solid'} size={'btn-sm'}>click me small</Button>
      ).toJSON();
      expect(button).toMatchSnapshot();
      expect(button.props.className).toEqual('btn btn-solid btn-sm init-caps');
    });

    it('should have disabled', () => {
      const button = renderer.create(
        <Button disabled >click me small</Button>
      ).toJSON();
      expect(button).toMatchSnapshot();
      expect(button.props.className).toEqual('btn btn-outline init-caps');
    });
  });
});
