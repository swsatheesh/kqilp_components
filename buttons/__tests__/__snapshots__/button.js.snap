// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Button /> should have a large size 1`] = `
<button
  className="btn btn-solid btn-lg init-caps"
  disabled={false}
  onClick={undefined}
>
  click me large
</button>
`;

exports[`component <Button /> should have a small size 1`] = `
<button
  className="btn btn-solid btn-sm init-caps"
  disabled={false}
  onClick={undefined}
>
  click me small
</button>
`;

exports[`component <Button /> should have disabled 1`] = `
<button
  className="btn btn-outline init-caps"
  disabled={true}
  onClick={undefined}
>
  click me small
</button>
`;

exports[`component <Button /> should render correctly 1`] = `
<button
  className="btn btn-outline init-caps"
  disabled={false}
  onClick={[Function]}
>
  click me
</button>
`;
